Add docs about universal CAN heartbeat (#1561)
wpilibsuite/frc-docs: source/docs/software/can-devices/can-addressing.rst

-  Support the minimum Broadcast message requirements as detailed in the Broadcast Messages section.
-  If controlling actuators, utilize a scheme to assure that the robot is issuing commands, is enabled, and is still present
-  Provide software library support for LabVIEW, C++, and Java or arrange with *FIRST*\ |reg| or FIRSTs Control System Partners to provide such interfaces.

Universal Heartbeat
-------------------

> The roboRIO provides a universal CAN heartbeat that any device on the bus can listen and react to. This heartbeat is sent every 20ms.

Reviewer: Are these three meant to be separate paragraphs or part of the same paragraph. They look a bit weird rendered on the docs since it's making them separate.

> The heartbeat has a full CAN ID of 0x01011840, which it NI manufacturer, RobotController type, Device ID 0 and API ID 0x062.

Reviewer 2: ```suggestion
The heartbeat contains: a full CAN ID of 0x01011840 (which is the NI manufacturer id), the RobotController type, Device ID 0 and API ID 0x062.
```

Author: Contains isn't the right word here. The Id is a descriptor and not contents.

Author: I fixed the wording up some.

It is an 8 byte CAN packet. The important byte in here is byte 5 (index 4). The layout is the following bitfield.

> RedAlliance : 1

Reviewer: We should turn this into a table. I'll push a commit for it.

Universal Heartbeat
-------------------

The roboRIO provides a universal CAN heartbeat that any device on the bus can listen and react to. This heartbeat is sent every 20ms. The heartbeat has a full CAN ID of ``0x01011840`` (which is the NI Manufacturer ID, RobotController type, Device ID 0 and API ID ``0x062``). It is an 8 byte CAN packet. The important byte in here is byte 5 (index 4). The layout is the following bitfield.

+-----------------+-------+
| Description     | Width |
+=================+=======+
| RedAlliance     | 1     |
+-----------------+-------+
| Enabled         | 1     |
+-----------------+-------+

> | Autonomous      | 1     |

Author: You forgot to put the test bit in the table.

Author: Also, I think this part specifically should stay how I had it. How I had it is how a bit field is designed in C, so that's really how people that understand bitfields would expect it to look.

Reviewer: Bitfields aren't shown in a table normally? You basically had a table previously... just without the table formatting.

Author: Not normally no. What I had is the actual C syntax for a bitfield which is pretty ubiquitous.

Reviewer: Are the previous examples in this article bitfields? Should they no longer be tables?

Reviewer: I also do see other examples of bitfields in tables from a brief google search

Author: The previous items are not bitfields, so those should be tables still. I guess a table is fine. But you do need to add in the item you missed. Otherwise its just wrong.
